tests/Aspire.Hosting.Redis.Tests/RedisFunctionalTests.cs

```diff
@@ -0,0 +1,227 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Components.Common.Tests;
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Utils;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using StackExchange.Redis;
+using Xunit;
+using Xunit.Sdk;
+
+namespace Aspire.Hosting.Redis.Tests;
+
+public class RedisFunctionalTests
+{
+    [Fact]
+    [RequiresDocker]
+    public async Task VerifyRedisResource()
+    {
+        var builder = TestDistributedApplicationBuilder.Create();
+
+        var redis = builder.AddRedis("redis");
+
+        using var app = builder.Build();
+
+        await app.StartAsync();
+
+        var hb = Host.CreateApplicationBuilder();
+
+        hb.Configuration.AddInMemoryCollection(new Dictionary<string, string?>
+        {
+            [$"ConnectionStrings:{redis.Resource.Name}"] = await redis.Resource.GetConnectionStringAsync()
+        });
+
+        hb.AddRedisClient(redis.Resource.Name);
+
+        using var host = hb.Build();
+
+        await host.StartAsync();
+
+        var redisClient = host.Services.GetRequiredService<IConnectionMultiplexer>();
+
+        var db = redisClient.GetDatabase();
+
+        await db.StringSetAsync("key", "value");
+
+        var value = await db.StringGetAsync("key");
+
+        Assert.Equal("value", value);
+    }
+
+    [Fact(Skip = "Debugging")]
```

Comment:
Seems like this "Skip"s should be removed.

Reply:
I was trying to debug why the tests were not working on the CI. Ankit made it clear now (docker registry throttling).